indico/htdocs/js/indico/jquery/permissions_widget.js

```diff
@@ -0,0 +1,327 @@
+            permissions.forEach(function(item) {
+                $permissionsList.append($('<li>', {class: 'i-label bold ' + permissionClasses[item]}).append(item));
+            });
+            $permissions.append(this._renderPermissionsButtons(principal, permissions));
+            return $permissions;
+        },
+        _renderPermissionsButtons: function(principal, permissions) {
+            var $buttonsGroup = $('<div>', {class: 'group flexrow'});
+            if (principal._type !== 'DefaultEntry') {
+                $buttonsGroup.append(this._renderEditBtn(principal, permissions), this._renderDeleteBtn(principal));
+                return $buttonsGroup;
+            }
+        },
+        _renderEditBtn: function(principal, permissions) {
+            var $permissionsEditBtn;
+            if (principal._type === 'IPNetworkGroup') {
+                $permissionsEditBtn = $('<button>', {
+                    type: 'button',
+                    class: 'i-button text-color borderless icon-only icon-edit disabled',
+                    title: $T('IP Networks can only have access permission')
+                });
+            } else {
+                $permissionsEditBtn = $('<button>', {
+                    'type': 'button',
+                    'class': 'i-button text-color borderless icon-only icon-edit',
+                    'data-href': build_url(Indico.Urls.EventPermissions, {confId: this.options.event_id}),
+                    'data-title': $T.gettext('Assign Permissions'),
+                    'data-ajax-dialog': '',
+                    'data-params': JSON.stringify({principal: JSON.stringify(principal), permissions: permissions})
+                });
+            }
+            return $permissionsEditBtn;
+        },
+        _renderDeleteBtn: function(principal) {
+            var self = this;
+            return $('<button>', {
+                'type': 'button',
+                'class': 'i-button text-color borderless icon-only icon-remove',
+                'data-principal': JSON.stringify(principal)
+            }).on('click', function() {
+                var $this = $(this);
+                var title = $T.gettext("Delete entry '{0}'".format(principal.name || principal.id));
+                var message = $T.gettext("Are you sure you want to permanently delete this entry?");
+                confirmPrompt(message, title).then(function() {
+                    self._updateItem($this.data('principal'), []);
+                });
+            });
+        },
+        _renderItem: function(item) {
+            var $item = $('<li>', {class: 'flexrow f-a-center'});
+            var principal = item[0];
+            var permissions = item[1];
+            $item.append(this._renderLabel(principal));
+            $item.append(this._renderPermissions(principal, permissions));
+            $item.toggleClass('disabled ' + principal.id, principal._type === 'DefaultEntry');
```

Comment:
In this case it's just for default entries, so the ids are `category-managers` and `anonymous`. I'm adding them as class because it later allows me to select the anonymous entry to change its permissions whenever the protection mode is changed.